src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -57,8 +63,18 @@ public EditCommand parse(String args) throws ParseException {
         if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
             editPersonDescriptor.setAddress(ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get()));
         }
-        if (argMultimap.getValue(PREFIX_USERTYPE).isPresent()) {
-            editPersonDescriptor.setUserType(ParserUtil.parseUserType(argMultimap.getValue(PREFIX_USERTYPE).get()));
+        // set the User as a Buyer if he/she has a preference
+        if (argMultimap.getValue(PREFIX_PREFERENCE).isPresent()) {
+            editPersonDescriptor.setUserType(UserTypeUtil.createBuyer());
+            editPersonDescriptor.setPreference(ParserUtil.parsePreference(
+                    argMultimap.getValue(PREFIX_PREFERENCE).get()));
+            editPersonDescriptor.clearProperties();
```

Comment:
I'm not sure if `clearProperties()` and `clearPreference()` is necessary

Reply:
yeah since a client can only have a `property` or `preference`, we'd need to clear the other out? if the edit is to make the client have a `property`, then clear his `preferences` & vice versa

Reply:
But entering this if block implies that the user has input only `Preference` and no `Property`. So I don't see how the `editPersonDescriptor` would have any `Property` to clear.